kong/tools/queue.lua

```diff
@@ -100,6 +100,95 @@ local function make_queue_key(name)
 end
 
 
+local function internal_remaining_capacity(self)
+  local remaining_entries = self.max_entries - self:count()
+
+  -- we enqueue entries one by one,
+  -- so it is impossible to have a negative value
+  assert(remaining_entries >= 0, "queue should not be over capacity")
+
+  if self.max_bytes == nil then
+    return remaining_entries
+  end
+
+  local remaining_bytes = self.max_bytes - self.bytes_queued
+
+  -- we check remaining_bytes before enqueuing an entry,
+  -- so it is impossible to have a negative value
+  assert(remaining_bytes >= 0, "queue should not be over capacity")
+
+  return remaining_entries, remaining_bytes
+end
+
+
+local function internal_is_reaching_max_entries(self)
+  -- `()` is used to get the first return value only
+  return (internal_remaining_capacity(self)) == 0
+end
+
+
+local function internal_will_exceed_max_entries(self)
+   -- `()` is used to get the first return value only
+  return (internal_remaining_capacity(self)) - 1 < 0
+end
+
+
+local function internal_is_entry_too_large(self, entry)
+  local max_bytes = self.max_bytes
+
+  if max_bytes == nil then
+    return false
+  end
+
+  if type(entry) ~= "string" then
+    -- handle non-string entry, including `nil`
+    return false
+  end
+
+  return #entry > max_bytes
+end
+
+
+local function internal_is_reaching_max_bytes(self)
+  if self.max_bytes == nil then
+    return false
+  end
+
+  local _, remaining_bytes = internal_remaining_capacity(self)
+  assert(type(remaining_bytes) == "number", "remaining_bytes must be a number")
+  return remaining_bytes == 0
+end
+
+
+local function internal_will_exceed_max_bytes(self, entry)
+  if self.max_bytes == nil then
+    return false
+  end
+
+  if type(entry) ~= "string" then
+    -- handle non-string entry, including `nil`
+    return false
+  end
+
+  local _, remaining_bytes = internal_remaining_capacity(self)
+  assert(type(remaining_bytes) == "number", "remaining_bytes must be a number")
+  return #entry > remaining_bytes
+end
+
+
+local function internal_is_full(self)
+  return internal_is_reaching_max_entries(self) or internal_is_reaching_max_bytes(self)
+end
+
+
+local function internal_can_enqueue(self, entry)
+  return internal_is_full(self)                       or
+         internal_is_entry_too_large(self, entry)     or
+         internal_will_exceed_max_entries(self)       or
+         internal_will_exceed_max_bytes(self, entry)
```

Comment:
It seems that either the logic needs to be reversed or the other functions are not correctly named now.
```suggestion
  return not (internal_is_full(self)                       or
         internal_is_entry_too_large(self, entry)     or
         internal_will_exceed_max_entries(self)       or
         internal_will_exceed_max_bytes(self, entry))
```

Reply:
fixed